src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let connector = separator;
+
+    if (separator === null) {
+      connector = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== this.length - 1 && this.length !== 0) {
```

Comment:
```suggestion
      if (i !== this.length - 1) {
```
If this.length = 0, you can't iterate through it.

Reply:
Fixed